test/spec/exerciseResult/exerciseResult.spec.js

```diff
@@ -353,4 +353,98 @@ describe('testing service "ExerciseResult":', function () {
             expect(exerciseResult.avgTimePerQuestion).toBe(expectedResult);
         });
     });
+
+    describe('test save exam isCompleted when all sections are completed',function(){
+
+        it('when section is not complete, then examResult.isComplete = false', function(){
+            var exerciseId = 4;
+            var examSectionsNum = 5;
+            var examId = 45;
+            var exerciseResult = actions.getExerciseResult(ExerciseTypeEnum.SECTION.enum, exerciseId, examId, examSectionsNum);
+            exerciseResult.$save();
+            $rootScope.$digest();
+            var examResult = actions.getExamResult(examId);
+            expect(examResult.isComplete).toEqual(false);
+        });
+
+        it('when section is complete, but other section not started, then examResult.isComplete = false', function(){
+            var exerciseId = 4;
+            var examSectionsNum = 5;
+            var examId = 45;
+            var exerciseResult = actions.getExerciseResult(ExerciseTypeEnum.SECTION.enum, exerciseId, examId, examSectionsNum);
+            exerciseResult.isComplete = true;
+            exerciseResult.$save();
+            $rootScope.$digest();
+            var examResult = actions.getExamResult(examId);
+            expect(examResult.isComplete).toEqual(false);
+        });
+
+        it('when section is complete, and other section are started but not completed, then examResult.isComplete = false', function(){
+            var exerciseId = 4;
+            var examSectionsNum = 5;
+            var examId = 45;
+            var sectionResultsObj = {
```

Comment:
the keys are like section id and ref section guid, we care only on the number of section results